### DISCUSSION TITLE
Add the static margin as a constraint

### DISCUSSION THREAD

**Discussion by friedenhe** [mdolab/dafoam#400] [General]
> I am learning to use DaFoam to optimize a flying wing, I need to add the longitudinal stability as a constraint which was calculated as the moment and lift derivatives with respect to the angle-of-attack. how to add the calculation the derivatives in DaFoam, could anyone please give some tips or tutorials? thanks.

@zhuwenguo, I moved your question from the DAFoam website repo to here.

**Comment by friedenhe** [comment]
We just added a script to optimize the NACA0012 airfoil with a static stability constraint. Check [here](https://github.com/DAFoam/tutorials/blob/main/NACA0012_Airfoil/incompressible/runScript_Stability.py)

**Reply by zhuwenguo** [reply]
Thank you!
